From 200112bc3f005bb5c45f964a2f0df5901125d9e9 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:09:54 -0400
Subject: [usb] pci-quirks: Prevent Sony VAIO t-series from switching usb ports

Message-id: <1410887463-20674-523-git-send-email-dzickus@redhat.com>
Patchwork-id: 95654
O-Subject: [RHEL7 PATCH 522/591] usb: pci-quirks: Prevent Sony VAIO t-series from switching usb ports
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit b38f09ccc3fd453180e96273bf3f34083c30809a
Author: Mathias Nyman <mathias.nyman@linux.intel.com>
Date:   Wed May 28 23:18:35 2014 +0300

    usb: pci-quirks: Prevent Sony VAIO t-series from switching usb ports

    Sony VAIO t-series machines are not capable of switching usb2 ports over
    from Intel EHCI to xHCI controller. If tried the USB2 port will be left
    unconnected and unusable.

    This patch should be backported to stable kernels as old as 3.12,
    that contain the commit 26b76798e0507429506b93cd49f8c4cfdab06896
    "Intel xhci: refactor EHCI/xHCI port switching"

    Cc: stable <stable@vger.kernel.org> # 3.12
    Reported-by: Jorge <xxopxe@gmail.com>
    Tested-by: Jorge <xxopxe@gmail.com>
    Signed-off-by: Mathias Nyman <mathias.nyman@linux.intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/pci-quirks.c b/drivers/usb/host/pci-quirks.c
index 00661d3..4a6d3dd 100644
--- a/drivers/usb/host/pci-quirks.c
+++ b/drivers/usb/host/pci-quirks.c
@@ -847,6 +847,13 @@ void usb_enable_intel_xhci_ports(struct pci_dev *xhci_pdev)
  bool  ehci_found = false;
  struct pci_dev *companion = NULL;
 
+ /* Sony VAIO t-series with subsystem device ID 90a8 is not capable of
+  * switching ports from EHCI to xHCI
+  */
+ if (xhci_pdev->subsystem_vendor == PCI_VENDOR_ID_SONY &&
+     xhci_pdev->subsystem_device == 0x90a8)
+  return;
+
  /* make sure an intel EHCI controller exists */
  for_each_pci_dev(companion) {
   if (companion->class == PCI_CLASS_SERIAL_USB_EHCI &&
-- 
1.7.1